apps/prairielearn/src/middlewares/authzCourseOrInstance.ts

```diff
@@ -552,9 +553,8 @@ export async function authzCourseOrInstance(req: Request, res: Response) {
     // are required.
     if (effectiveAuthResult === null) {
       return {
-        authzData: {
+        authzData: withBrand<PlainAuthzData>({
           user: effectiveUserData ? effectiveUserData.user : authnAuthzData.user,
-          is_administrator: false,
```

Comment:
TypeScript caught this extra property once I added `withBrand(...)` here.

Reply:
I don't see how they could have been, but let me know if I'm missing something.

This is in the middleware, and specifically in a `run(...)` block. That returns this `authzData` into a `effectiveAuthzData` object, but that object as a whole never leaves this middleware. Instead, we pick off the specific properties we need and use that to construct `res.locals.authz_data` from scratch.

Does that analysis sound correct to you?